### ISSUE TITLE
Is it possible to make computer_use_demo's bash session visible and interactive on the screen?

### ISSUE THREAD

**Issue by asdf2adsfad** [anthropics/claude-quickstarts#112]
As title, this would be extremely useful for saving screenshot tokens and increase speed

Also this would enable Claude to use tools like nmtui nano top etc.

**Comment by chrisgorgo** [comment]
You can disable BashTool and prompt the model to open visual terminal and use it for issuing commands. It wouldn't save tokens though.

**Comment by asdf2adsfad** [comment]
> You can disable BashTool and prompt the model to open visual terminal and use it for issuing commands. It wouldn't save tokens though.

Yes that's true, but as far as I can tell, it seemed that interacting with the terminal window is really slow comparing to direct interaction with shell tools. I used edited the bash tool to create a web interface, and it is really useful (yet not perfect and ugly)

**Comment by p-i-** [comment]
The problem is the bash_tool doesn't use a running bash session. It probably executes a single bash command in a subshell. It does dump the bash-commands to the convo, sure would be hella nice if it multi-lined it. Adding that one to my kanban.
